src/odemis/acq/stream/_sync.py

```diff
@@ -773,6 +778,143 @@ def _stopLeeches(self):
+        :param raw_data: acquired data of SEM stream
+        :param px_idx: (tuple of int) pixel index: y, x
+        :param rep: (tuple of int) repetition frame/ size of entire frame
+        :param pol_idx: (int) polarisation index related to name as defined in pos_polarizations variable
+        uses fuzzing
+        """
+        tile_shape = raw_data.shape
+
+        if pol_idx > len(self._live_data[n]) - 1:
+            # New polarization => new DataArray
+            md = raw_data[0].metadata.copy()
+            center, pxs = self._get_center_pxs(rep, tile_shape , raw_data)
+            md.update({MD_POS: center,
+                       MD_PIXEL_SIZE: pxs})
+            da = model.DataArray(numpy.zeros(shape=rep[::-1] * numpy.array(tile_shape), dtype=raw_data.dtype), md)
+            self._live_data[n].append(da)
+            self._acq_mask = numpy.zeros(shape=rep[::-1] * numpy.array(tile_shape), dtype=numpy.bool)
+
+        self._acq_mask[px_idx[0] * tile_shape[0]:(px_idx[0] + 1) * tile_shape[0],
+                       px_idx[1] * tile_shape[1]:(px_idx[1] + 1) * tile_shape[1]] = True
+        self._live_data[n][pol_idx][
+                       px_idx[0] * tile_shape[0]:(px_idx[0] + 1) * tile_shape[0],
+                       px_idx[1] * tile_shape[1]:(px_idx[1] + 1) * tile_shape[1]] = raw_data
+
+    def _assembleLiveData2D(self, n, raw_data, px_idx, rep, pol_idx):
+        """
+        :param n: (int) number of the current stream
+        :param raw_data: acquired data of stream
+        :param px_idx: (tuple of int) pixel index: y, x
+        :param rep: (tuple of int) repetition frame/ size of entire frame
+        :param pol_idx: (int) polarisation index related to name as defined in pos_polarizations variable
+        """
+        tile_shape = raw_data.shape
+
+        if pol_idx > len(self._live_data[n]) - 1:
+            # New polarization => new DataArray
+            md = raw_data[0].metadata.copy()
+            center, pxs = self._get_center_pxs(rep, (1,1) , raw_data)
+            md.update({MD_POS: center,
+                       MD_PIXEL_SIZE: pxs})
+            da = model.DataArray(numpy.zeros(shape=rep[::-1], dtype=raw_data.dtype), md)
+            self._live_data[n].append(da)
+            self._acq_mask = numpy.zeros(rep[::-1], dtype=numpy.bool)
+
+        self._acq_mask[px_idx[0]: px_idx[0] + tile_shape[0],
+                       px_idx[1]: px_idx[1] + tile_shape[1]] = True
+        self._live_data[n][pol_idx][
+                           px_idx[0]: px_idx[0] + tile_shape[0],
+                           px_idx[1]: px_idx[1] + tile_shape[1]] = raw_data
+
+    def _assembleFinalData(self, data, pol_idx, pos_polarizations, stream_idx):
+        '''
+        :param data: all acquired data of the streams
+        :param pol_idx: (int) polarisation index related to name as defined in pos_polarizations variable
+        :param pos_polarizations: (list of str) number of possible polarization possibilities
```

Comment:
pos_polarizations we need to not average if there are not multiple polarisations.
pol_idx is very nice if stuff goes wrong for debugging. Since it does not make the function unreadable or messy I propose to leave it in place.

Reply:
ok remove